internal/watch/watch.go

```diff
@@ -288,3 +289,95 @@ func renderLogs(ctx context.Context, alloc *api.AllocationStatus) {
 		)
 	}
 }
+
+func MachinesChecks(ctx context.Context) error {
```

Comment:
We may want to adjust this logic a bit.  

So instead of having a single function that monitors health checks for all machines, we should consider having a single function that takes a machine and monitors its associated health checks.   There may be times where we need to be explicit about which Machines we want to monitor, for example, we only care about PG machines within the context of a PG app.  We can ignore any non-PG Machines that may have been provisioned.  

I think this will make it a bit more flexible as we can leverage the same function for restarts/updates and provisions.  Restarts/updates will need to monitor machine independently one-by-one, while post-provision we could look to monitor Machines concurrently.